uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-widget/leave-widget-balance.component.js

```diff
@@ -0,0 +1,106 @@
+/* eslint-env amd */
+
+define([
+  'common/lodash',
+  'leave-absences/shared/modules/components',
+  'leave-absences/shared/models/entitlement.model'
+], function (_, components) {
+  components.component('leaveWidgetBalance', {
+    bindings: {
+      absenceTypes: '<',
+      contactId: '<',
+      currentAbsencePeriod: '<'
+    },
+    controller: leaveWidgetBalanceController,
+    controllerAs: 'leaveWidgetBalance',
+    templateUrl: ['shared-settings', function (sharedSettings) {
+      return sharedSettings.sharedPathTpl + 'components/leave-widget/leave-widget-balance.html';
+    }]
+  });
+
+  leaveWidgetBalanceController.$inject = ['$scope', 'Entitlement'];
+
+  function leaveWidgetBalanceController ($scope, Entitlement) {
+    var entitlements;
+    var vm = this;
+    vm.$onChanges = $onChanges;
+
+    /**
+     * Initializes the component by emiting a child is loading event.
+     */
+    (function init () {
+      $scope.$emit('LeaveWidget::childIsLoading');
+    })();
+
+    /**
+     * Implements the $onChanges controller method. It watches for changes in
+     * the component bindings.
+     *
+     * @param {Object} change - the change object with the current and previous
+     * value of the binding.
+     */
+    function $onChanges (change) {
+      if (areBindingsReady()) {
+        loadDependencies();
+      }
+    }
+
+    /**
+     * Returns true if all bindings are ready and can be used by the component.
+     *
+     * @return {Boolean}
+     */
+    function areBindingsReady () {
+      return vm.absenceTypes && vm.currentAbsencePeriod && vm.contactId;
+    }
+
+    /**
+     * Loads all the component dependencies (entitlements in this case) and
+     * emits a child is ready event.
+     */
+    function loadDependencies () {
+      loadEntitlements()
+      .then(function () {
```

Comment:
I think in this case .then can go right after loadEntitlements()

```js
loadEntitlements().then(function () {
//...
```

up to you.